### PR TITLE
EE-1140: remove grpc layer

### PR DESCRIPTION
**Pull request by EdHastingsCasperLabs**
This PR removes the legacy grpc layer.

https://casperlabs.atlassian.net/browse/EE-1140

Basically mechanical, except for the concurrent_executor test deprecation.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -153,7 +152,6 @@ check: \
 	build \
 	check-format \
 	lint \
-	audit \
```
Comment (Fraser999) [inline Makefile:156]: I missed this when I committed the workaround - I only changed the drone file.

If you want to leave audit in with just that single warning ignored, you can change the audit target to do
```
$(CARGO) audit --ignore RUSTSEC-2020-0123
```

**Review by Fraser999 — APPROVED** [review]
Great to see this one land!

One minor non-blocker, but otherwise looks grand.

File: Makefile (hunk 1 above)
Comment (EdHastingsCasperLabs) [reply Makefile:156]: k

**Review by mpapierski — APPROVED** [review]

**Review by henrytill — APPROVED** [review]

**Pull request merged**